File: APPManage/AM.Model/tower_crane.cs
/**  版本信息模板在安装目录下，可自行修改。
* tower_crane.cs
*
* 功 能： N/A
* 类 名： tower_crane
*
* Ver    变更日期             负责人  变更内容
* ───────────────────────────────────
* V0.01  2016/8/12 22:32:20   N/A    初版
*
* Copyright (c) 2012 Maticsoft Corporation. All rights reserved.
*┌──────────────────────────────────┐
*│　此技术信息为本公司机密信息，未经本公司书面同意禁止向第三方披露．　│
*│　版权所有：动软卓越（北京）科技有限公司　　　　　　　　　　　　　　│
*└──────────────────────────────────┘
*/
using System;
namespace AM.Model
{
	/// <summary>
	/// tower_crane:实体类(属性说明自动提取数据库字段的描述信息)
	/// </summary>
	[Serializable]
	public partial class tower_crane
	{
		public tower_crane()
		{}
		#region Model
		private string _guid;
		private string _userid;
		private DateTime? _createdate;
		private string _gcmc;
		private string _sydw;
		private string _sgdd;
		private string _jldw;
		private string _azdw;
		private string _azfzr;
		private string _sbmc;
		private string _sbxh;
		private string _zzdw;
		private string _azgzrq;
		private string _ccbh;
		private string _zzxkzbh;
		private string _ccrq;
		private string _babh;
		private string _zdedqzl;
		private string _edlj;
		private string _xczdbfsd;
		private string _bzjggsl;
		private string _jysazgd;
		private string _zdfd;
		private string _yxdlgd;
		private string _azfzds;
		private string _dydfqgd;
		private string _jylb;
		private string _zysbmc1;
		private string _zysbxh1;
		private string _zysbbh1;
		private string _zysbzt1;
		private string _zysbmc2;
		private string _zysbxh2;
		private string _zysbbh2;
		private string _zysbzt2;
		private string _zysbmc3;
		private string _zysbxh3;
		private string _zysbbh3;
		private string _zysbzt3;
		private string _zysbmc4;
		private string _zysbxh4;
		private string _zysbbh4;
		private string _zysbzt4;
		private string _zysbmc5;
		private string _zysbxh5;
		private string _zysbbh5;
		private string _zysbzt5;
		private string _bzxmbhgs;
		private string _ybxmbhgs;
		/// <summary>
		/// 
		/// </summary>
		public string Guid
		{
			set{ _guid=value;}
			get{return _guid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string userid
		{
			set{ _userid=value;}
			get{return _userid;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime? CreateDate
		{
			set{ _createdate=value;}
			get{return _createdate;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string GCMC
		{
			set{ _gcmc=value;}
			get{return _gcmc;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SYDW
		{
			set{ _sydw=value;}
			get{return _sydw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SGDD
		{
			set{ _sgdd=value;}
			get{return _sgdd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string JLDW
		{
			set{ _jldw=value;}
			get{return _jldw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZDW
		{
			set{ _azdw=value;}
			get{return _azdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZFZR
		{
			set{ _azfzr=value;}
			get{return _azfzr;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SBMC
		{
			set{ _sbmc=value;}
			get{return _sbmc;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string SBXH
		{
			set{ _sbxh=value;}
			get{return _sbxh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZZDW
		{
			set{ _zzdw=value;}
			get{return _zzdw;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZGZRQ
		{
			set{ _azgzrq=value;}
			get{return _azgzrq;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string CCBH
		{
			set{ _ccbh=value;}
			get{return _ccbh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZZXKZBH
		{
			set{ _zzxkzbh=value;}
			get{return _zzxkzbh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string CCRQ
		{
			set{ _ccrq=value;}
			get{return _ccrq;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BABH
		{
			set{ _babh=value;}
			get{return _babh;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZDEDQZL
		{
			set{ _zdedqzl=value;}
			get{return _zdedqzl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string EDLJ
		{
			set{ _edlj=value;}
			get{return _edlj;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string XCZDBFSD
		{
			set{ _xczdbfsd=value;}
			get{return _xczdbfsd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BZJGGSL
		{
			set{ _bzjggsl=value;}
			get{return _bzjggsl;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string JYSAZGD
		{
			set{ _jysazgd=value;}
			get{return _jysazgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZDFD
		{
			set{ _zdfd=value;}
			get{return _zdfd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YXDLGD
		{
			set{ _yxdlgd=value;}
			get{return _yxdlgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string AZFZDS
		{
			set{ _azfzds=value;}
			get{return _azfzds;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string DYDFQGD
		{
			set{ _dydfqgd=value;}
			get{return _dydfqgd;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string JYLB
		{
			set{ _jylb=value;}
			get{return _jylb;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBMC1
		{
			set{ _zysbmc1=value;}
			get{return _zysbmc1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBXH1
		{
			set{ _zysbxh1=value;}
			get{return _zysbxh1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBBH1
		{
			set{ _zysbbh1=value;}
			get{return _zysbbh1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBZT1
		{
			set{ _zysbzt1=value;}
			get{return _zysbzt1;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBMC2
		{
			set{ _zysbmc2=value;}
			get{return _zysbmc2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBXH2
		{
			set{ _zysbxh2=value;}
			get{return _zysbxh2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBBH2
		{
			set{ _zysbbh2=value;}
			get{return _zysbbh2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBZT2
		{
			set{ _zysbzt2=value;}
			get{return _zysbzt2;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBMC3
		{
			set{ _zysbmc3=value;}
			get{return _zysbmc3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBXH3
		{
			set{ _zysbxh3=value;}
			get{return _zysbxh3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBBH3
		{
			set{ _zysbbh3=value;}
			get{return _zysbbh3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBZT3
		{
			set{ _zysbzt3=value;}
			get{return _zysbzt3;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBMC4
		{
			set{ _zysbmc4=value;}
			get{return _zysbmc4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBXH4
		{
			set{ _zysbxh4=value;}
			get{return _zysbxh4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBBH4
		{
			set{ _zysbbh4=value;}
			get{return _zysbbh4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBZT4
		{
			set{ _zysbzt4=value;}
			get{return _zysbzt4;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBMC5
		{
			set{ _zysbmc5=value;}
			get{return _zysbmc5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBXH5
		{
			set{ _zysbxh5=value;}
			get{return _zysbxh5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBBH5
		{
			set{ _zysbbh5=value;}
			get{return _zysbbh5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string ZYSBZT5
		{
			set{ _zysbzt5=value;}
			get{return _zysbzt5;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string BZXMBHGS
		{
			set{ _bzxmbhgs=value;}
			get{return _bzxmbhgs;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string YBXMBHGS
		{
			set{ _ybxmbhgs=value;}
			get{return _ybxmbhgs;}
		}
		#endregion Model

	}
}

File: AM/Scripts/appm/tower_crane.js
function testtc() {
    var isok = false;
    if ($("#gcname").val() == "") {
        alert("工程名称不能为空");
        $("#gcname").focus();
    } else {
        if ($("#sycompany").val() == "") {
            alert("使用单位不能为空");
            $("#sycompany").focus();
        } else {
            if ($("#sgaddress").val() == "") {
                alert("施工地点不能为空");
                $("#sgaddress").focus();
            } else {
                if ($("#jlcompany").val() == "") {
                    alert("监理单位不能为空");
                    $("#jlcompany").focus();
                } else {
                    if ($("#azcompany").val() == "") {
                        alert("安装单位不能为空");
                        $("#azcompany").focus();
                    } else {
                        if ($("#azpeople").val() == "") {
                            alert("安装负责人不能为空");
                            $("#azpeople").focus();
                        } else {
                            if ($("#sbname").val() == "") {
                                alert("设备名称不能为空");
                                $("#sbname").focus();
                            } else {
                                if ($("#type").val() == "") {
                                    alert("设备型号不能为空");
                                    $("#type").focus();
                                } else {
                                    if ($("#zzcompany").val() == "") {
                                        alert("制造单位不能为空");
                                        $("#zzcompany").focus();
                                    } else {
                                        if ($("#azdate").val() == "") {
                                            alert("安装告知日期不能为空");
                                            $("#azdate").focus();
                                        } else {
                                            if ($("#cccode").val() == "") {
                                                alert("出厂编号不能为空");
                                                $("#cccode").focus();
                                            } else {
                                                if ($("#xkzcode").val() == "") {
                                                    alert("制造许可证编号不能为空");
                                                    $("#xkzcode").focus();
                                                } else {
                                                    if ($("#ccdate").val() == "") {
                                                        alert("出厂日期不能为空");
                                                        $("#ccdate").focus();
                                                    } else {
                                                        if ($("#bacode").val() == "") {
                                                            alert("备案编号不能为空");
                                                            $("#bacode").focus();
                                                        } else {
                                                            if ($("#maxweight").val() == "") {
                                                                alert("最大额定起重量不能为空");
                                                                $("#maxweight").focus();
                                                            } else {
                                                                if ($("#edlj").val() == "") {
                                                                    alert("额定力矩不能为空");
                                                                    $("#edlj").focus();
                                                                } else {
                                                                    if ($("#maxspeed").val() == "") {
                                                                        alert("小车最大变幅速度不能为空");
                                                                        $("#maxspeed").focus();
                                                                    } else {
                                                                        if ($("#stnumber").val() == "") {
                                                                            alert("标准节规格/数量不能为空");
                                                                            $("#stnumber").focus();
                                                                        } else {
                                                                            if ($("#checkheight").val() == "") {
                                                                                alert("检验时安装高度不能为空");
                                                                                $("#checkheight").focus();
                                                                            } else {
                                                                                if ($("#maxrange").val() == "") {
                                                                                    alert("最大幅度/安装幅度不能为空");
                                                                                    $("#maxrange").focus();
                                                                                } else {
                                                                                    if ($("#yxheight").val() == "") {
                                                                                        alert("允许独立高度/允许自由端高度不能为空");
                                                                                        $("#yxheight").focus();
                                                                                    } else {
                                                                                        if ($("#checkspace").val() == "") {
                                                                                            alert("检验时安装附着道数/间距不能为空");
                                                                                            $("#checkspace").focus();
                                                                                        } else {
                                                                                            if ($("#wallheight").val() == "") {
                                                                                                alert("第一道附墙高度不能为空");
                                                                                                $("#wallheight").focus();
                                                                                            } else {
                                                                                                if ($("#checktype").val() == "") {
                                                                                                    alert("检验类别不能为空");
                                                                                                    $("#checktype").focus();
                                                                                                } else
                                                                                                    isok = true;
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    return isok;
}
String.prototype.trim = function () {

    return this.replace(/(^\s*)|(\s*$)/g, '');
}

function submit() {
    var check = testtc();
    if (check)
    {
    var userid = $("#userid").val();
    var objArr = $("input[type=text]")
    var mobjArr1 = $("input[mtype=1]");
    var json = {
        "tower_crane": [{
            "Guid": "",
            "CreateDate": "",
            "userid": userid,
            "GCMC": $(objArr[1]).val(),
            "SYDW": $(objArr[2]).val(),
            "SGDD": $(objArr[3]).val(),
            "JLDW": $(objArr[4]).val(),
            "AZDW": $(objArr[5]).val(),
            "AZFZR": $(objArr[6]).val(),
            "SBMC" : $(objArr[7]).val(),
            "SBXH": $(objArr[8]).val(),
            "ZZDW": $(objArr[9]).val(),
            "AZGZRQ": $(objArr[10]).val(),
            "CCBH": $(objArr[11]).val(),
            "ZZXKZBH": $(objArr[12]).val(),
            "CCRQ": $(objArr[13]).val(),
            "BABH": $(objArr[14]).val(),
            "ZDEDQZL": $(objArr[15]).val(),
            "EDLJ": $(objArr[16]).val(),
            "XCZDBFSD": $(objArr[17]).val(),
            "BZJGGSL": $(objArr[18]).val(),
            "JYSAZGD": $(objArr[19]).val(),
            "ZDFD": $(objArr[20]).val(),
            "YXDLGD": $(objArr[21]).val(),
            "AZFZDS": $(objArr[22]).val(),
            "DYDFQGD": $(objArr[23]).val(),
            "JYLB": $(objArr[24]).val(),
            "ZYSBMC1": $(mobjArr1[0]).val(),
            "ZYSBXH1": $(mobjArr1[1]).val(),
            "ZYSBBH1": $(mobjArr1[2]).val(),
            "ZYSBZT1": $(mobjArr1[3]).val(),
            "ZYSBMC2": $(mobjArr1[4]).val(),
            "ZYSBXH2": $(mobjArr1[5]).val(),
            "ZYSBBH2": $(mobjArr1[6]).val(),
            "ZYSBZT2": $(mobjArr1[7]).val(),
            "ZYSBMC3": $(mobjArr1[8]).val(),
            "ZYSBXH3": $(mobjArr1[9]).val(),
            "ZYSBBH3": $(mobjArr1[10]).val(),
            "ZYSBZT3": $(mobjArr1[11]).val(),
            "ZYSBMC4": $(mobjArr1[12]).val(),
            "ZYSBXH4": $(mobjArr1[13]).val(),
            "ZYSBBH4": $(mobjArr1[14]).val(),
            "ZYSBZT4": $(mobjArr1[15]).val(),
            "ZYSBMC5": $(mobjArr1[16]).val(),
            "ZYSBXH5" : $(mobjArr1[17]).val(),
            "ZYSBBH5": $(mobjArr1[18]).val(),
            "ZYSBZT5": $(mobjArr1[19]).val(),
            "BZXMBHGS": $(objArr[objArr.length-2]).val(),
            "YBXMBHGS": $(objArr[objArr.length - 1]).val(),
        }]
    };
    $.ajax({
        type: "post",
        url: "/AppM/addtower_crane",
        data: { obj: JSON.stringify(json) },
        success: function (data) {
            layer.msg("提交成功");
        }
    })
 
    }
}